lightning/src/ln/channelmanager.rs

```diff
@@ -7117,8 +7377,13 @@ where
 						// without the new monitor persisted - we'll end up right back here on
 						// restart.
 						let previous_channel_id = claimable_htlc.prev_hop.outpoint.to_channel_id();
-						if let Some(channel) = by_id.get_mut(&previous_channel_id) {
-							channel.claim_htlc_while_disconnected_dropping_mon_update(claimable_htlc.prev_hop.htlc_id, payment_preimage, &args.logger);
+						if let Some(peer_node_id) = id_to_peer.get(&previous_channel_id){
+							let peer_state_mutex = per_peer_state.get(peer_node_id).unwrap();
```

Comment:
Do we know this unwrap is safe?

Reply:
Yeah agree that this is a bit confusing :). Even the channel existence check below is actually redundant, as we're guaranteed to have the channel here if it's exists in the `id_to_peer` map. I can of course add an extra `if` clause here though if you think it makes the code less confusing :).

Another general problem here going forward though: I'm a little worried here what we're gonna do here once we remove the `id_to_peer` map. Ideally we should be using the `short_to_chan_info` map here instead. Gonna look more into if we're guaranteed to have all channels we can have `serialized` claims for in the `short_to_chan_info` map here, especially channels that's not fully closed but about to (ie. `is_usable` is false).